xla/hlo/translate/mhlo_to_hlo/mlir_hlo_to_hlo.cc

```diff
@@ -2851,12 +2852,14 @@ LogicalResult ExportXlaOp(CustomCallOp op, OpLoweringContext ctx) {
     return op.emitOpError()
            << "cannot export with more than one called computations";
 
-  // Custom call can be exported either with called computation or with layout
-  // attributes. The XlaBuilder API does not allow both.
-  if (!op.getCalledComputations().empty() && op.getOperandLayouts() &&
-      op.getResultLayouts()) {
-    return op.emitOpError() << "cannot export if both called computation and "
-                               "layouts are specified";
+  if (op.getCallTargetName().str().find(kCudnnFMHA) == std::string::npos) {
```

Comment:
Can you please add a test for this change?
When I read the code further down below, I do believe that it does not handle cases where a custom call has both a called computation and operand layout or result layout. You can see a check for called computations, and then an else if for operand layout and result layout.
So I believe that excluding a custom call with target kCudnnFMHA from the check here is not sufficient, you would also need to adapt the logic down below if you want to support having both called computation and result layout.
Also, is this something that needs to be done in this PR, or could it be separate?

Reply:
Thanks for the feedback, I double checked and it is right that if computation is needed, `CustomCallWithComputation` will be called and if layout is needed, `CustomCallWithLayout` will be called. `CustomCallWithComputation`  always used `nullopt` when calls into `builder->CustomCall`. I think I can try modifying `CustomCallWithComputation`  to accept layout. Let me create a separate PR for this.

Reply:
I think we still need to add these changes to make sure it allow both layouts and computation?

Reply:
If the other change lands first, then we don't need this. I would prefer not to merge this PR if we do not actually support a custom call with computation and layout constraint. But maybe I misunderstood something? Why does this custom call have layout constraints, is it maybe not actually needed? If we merge the PR like this, we are essentially saying: ignore the layout constraints (as it would be handled by the code for custom calls with just a computation).

Reply:
Yes, I think the other PR should go in first, I remember if no layout constraints are specified, then layouts may get altered by XLA and cuDNN has a specific requirement for the layout.